app/models/manageiq/providers/ansible_runner_workflow.rb

```diff
@@ -128,4 +139,23 @@ def load_transitions
       :error        => {'*'                => '*'}
     }
   end
+
+  private
+
+  def checkout_git_repository
+    return unless options[:configuration_script_source_id]
+
+    css = ManageIQ::Providers::EmbeddedAnsible::AutomationManager::ConfigurationScriptSource.find(options[:configuration_script_source_id])
+    options[:git_checkout_dir] = Dir.mktmpdir("ansible-runner-git")
+    save!
+    _log.info("Checking out git repository to #{options[:git_checkout_dir].inspect}...")
+    css.checkout_git_repository(options[:git_checkout_dir])
+  end
+
+  def cleanup_git_repository
+    return unless options[:git_checkout_dir]
```

Comment:
Are we ever expecting someone to set this external to the workflow? If not, I would make it even more explicit that it's something we're going to remove by calling it `git_checkout_tempdir` or something.

Reply:
+1 for tempdir in the name.